src/wallet/coinselection.cpp

```diff
@@ -298,3 +298,35 @@ bool KnapsackSolver(const CAmount& nTargetValue, std::vector<CInputCoin>& vCoins
 
     return true;
 }
+
+/******************************************************************************
+
+ OutputGroup
+
+ ******************************************************************************/
+
+void OutputGroup::Insert(const CInputCoin& output, int depth, bool from_me, size_t ancestors, size_t descendants) {
+    m_outputs.push_back(output);
+    m_from_me &= from_me;
+    m_value += output.effective_value;
+    m_depth = std::min(m_depth, depth);
+    m_ancestors = std::max(m_ancestors, ancestors);
+    m_descendants = std::max(m_descendants, descendants);
```

Comment:
Can you explain why this is taking `std::max`?  I think the correct calculation would involve taking the set of ancestors or descendants of the group and counting the unique entries.

Reply:
The `descendants` value is already a complete count of all descendants starting from the top ancestor of the transaction in question. Here we only really care about `who has the most`.

Reply:
I believe this calculation may be fine for descendants (if I'm correctly understanding what the descendants variable represents -- it could use a code comment explaining exactly what it is, I think the `descendants` variable is the max number of descendants of any ancestor of a given coin?), however I think it is incorrect for ancestors.

I believe the ancestor count is the actual number of ancestors a given coin has.  If you just take the max across all coins in an output group, then if you have e.g. 30 2-ancestor coins in an output group, you may treat them as a 2-ancestor output group, when in fact you have 60 ancestors in any transaction that includes the group.

(For descendants, I think this works out fine, because all we're doing by creating a new transaction is adding one more descendant to every ancestor, so considering the max value is sufficient to try to avoid hitting mempool chain limits.)

I understand that the existing logic may be imprecise and hacky around this, but I believe that combining all the coins into an output group exacerbates this issue.

Regardless of what we end up doing here though I think the code could use a comment explaining the logic, as this is non-obvious.

Reply:
> I believe the ancestor count is the actual number of ancestors a given coin has. If you just take the max across all coins in an output group, then if you have e.g. 30 2-ancestor coins in an output group, you may treat them as a 2-ancestor output group, when in fact you have 60 ancestors in any transaction that includes the group.

Aha! This is a very good point. I will think about a remedy for this post-merge. For now, I added a comment explaining how it works and how it can be improved.